Harmony/Tools/AccessTools.cs

```diff
@@ -1687,11 +1692,18 @@ public static object MakeDeepCopy(object source, Type resultType, Func<string, T
 
 			if (type.IsGenericType && resultType.IsGenericType)
 			{
-				var addOperation = FirstMethod(resultType, m => m.Name == "Add" && m.GetParameters().Count() == 1);
-				if (addOperation is object)
+				if (!handlerCache.TryGetValue(resultType, out FastInvokeHandler addInvoker))
+				{
+					var addOperation = FirstMethod(resultType, m => m.Name == "Add" && m.GetParameters().Count() == 1);
```

Comment:
Sneak in another minor optimization here: use `Count` rather than `Count()`

Reply:
Or rather, `Length` since `GetParameters()` returns an array.

Reply:
It is legacy code, I will make the change.